.github/workflows/test-deploy.yml

```diff
@@ -0,0 +1,71 @@
+name: Test superset deployment
+
+on: push
+
+jobs:
+  test_local:
+    name: Superset local deployment
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+          cache: 'npm'
+      - run: sudo apt-get update && sudo apt-get install curl
+      - run: npm ci
+      - run: ./bin/dev --version
+      - run: ./bin/dev config:set registry_host registry.dev.architect.io
+      - run: ./bin/dev config:set api_host https://api.dev.architect.io
+      - run: ./bin/dev config:set log_level debug
+      - run: ./bin/dev login -e ${{ secrets.ARCHITECT_EMAIL }} -p ${{ secrets.ARCHITECT_PASSWORD }}
+      - run: docker volume create volume-key
+      - run: ./bin/dev link examples/hello-world/architect.yml
+      - run: ./bin/dev dev -e superset test/mocks/superset/architect.yml --browser=false -s param_unset=test -s world_text=Architect > ./tmp.txt 2>&1 &
+      - run: sh scripts/server_online.sh
+      - name: Check that the app is accessible locally
+        run: curl --fail -S -I https://hello.localhost.architect.sh
+      - name: Check that the app returns the expected response
+        run: curl --fail -S https://hello.localhost.architect.sh | grep "Hello Architect!"
+      - name: Run a local task
+        run: ./bin/dev task superset curler-build --local -e superset
+
+  test_remote:
+    name: Superset remote deployment
+    if: github.ref_name == 'main' || github.ref_name == 'rc'
```

Comment:
Lets make this only for main

Reply:
Also is ref_name the branch you'd be merging into or the current branch?

Reply:
That's the branch that was pushed to. We use the same logic in the publish workflow

https://github.com/architect-team/architect-cli/blob/rc/.github/workflows/publish.yml#L58

Reply:
We want these tests to run when we open a pr from `rc` into `main`. The idea being that we don't allow anything to go into mainline if the api hasn't been updated. Sorry if that wasn't clear. 

So if we open a pr on `rc` then I'd imagine `github.ref_name` would be `rc`. According to their docs `The short ref name of the branch or tag that triggered the workflow run.`.

I think this is a scenario where we actually want to use `pull_request_target` as the trigger. You'd need to confirm, but that might make things awkward since we want the remote test to only run on prs to main, but the local tests to run on all prs.

Reply:
Oh looks like you can target specific events:
https://docs.github.com/en/actions/learn-github-actions/contexts#example-usage-of-the-github-context

Reply:
Wouldn't the issue in using `pull_request_target` be the following from the [docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target)? So using that wouldn't use any changes that we want to merge

> This event runs in the context of the base of the pull request, rather than in the context of the merge commit, as the pull_request event does